checker/checker.go

```diff
@@ -567,7 +598,7 @@ func (v *visitor) ConditionalNode(node *ast.ConditionalNode) reflect.Type {
 	if t1 == nil && t2 == nil {
 		return nilType
 	}
-	if t1.AssignableTo(t2) {
+	if t1 != nil && t1.AssignableTo(t2) {
```

Comment:
due to previous cases t1 != nil  should never happen, why did you modify this ?

Reply:
indeed, my mistake !